python linting `make pre-commit`

### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of polars.


### Issue Description

Hey @stinodego -- 

Really like your new python linting. Looking high quality!

I'm getting an error using the `make pre-commit` locally to check linting.

Running in docker with a base image `python:3.10`

Is it obvious to you that I'm doing something wrong?

```bash
make pre-commit
venv/bin/isort .
Skipped 1 files
venv/bin/black .
All done! ✨ 🍰 ✨
110 files left unchanged.
venv/bin/blackdoc .
All done! ✨ 🍰 ✨
129 files left unchanged.
venv/bin/pyupgrade --py37-plus `find polars/ tests/ -name "*.py" -type f`
venv/bin/mypy
venv/lib/python3.10/site-packages/numpy/__init__.pyi:636: error: Positional-only parameters are only supported in Python 3.8 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
make: *** [Makefile:35: pre-commit] Error 2
```

### Reproducible Example

```python
# Start from a 3.10 image
docker run -v /home/r/lab/polars/:/polars python:3.10 sleep 9999
docker exec -it ... bash
# wants rustc, let's add it
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh 
make pre-commit
```


### Expected Behavior

success on make `pre-commit`

### Installed Versions

I must be missing something on this too? `pl.show_versions()` errors?

Here's a readable output..

```bash
root@eb7b441203cb:/polars/py-polars# pip3 list
Package    Version
---------- -------
pip        22.2.2
polars     0.14.13
setuptools 63.2.0
wheel      0.37.1
root@eb7b441203cb:/polars/py-polars# python3 --version
Python 3.10.7
root@eb7b441203cb:/polars/py-polars# uname -a
Linux eb7b441203cb 5.15.0-46-generic #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022 x86_64 GNU/Linux

```

<details>
```
>>> import polars as pl
>>> pl.show_versions()
---Version info---
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/polars/py-polars/polars/show_versions.py", line 38, in show_versions
    print(f"Polars: {version()}")
NameError: name 'version' is not defined


 ```
</details>

--- a/.github/workflows/test-python.yaml
+++ b/.github/workflows/test-python.yaml
@@ -21,9 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # TODO: Change '3.10.6' to '3.10' after updating mypy version
-        # See issue: https://github.com/pola-rs/polars/issues/4957
-        python-version: ['3.7', '3.10.6']
+        python-version: ['3.7', '3.10']
 
     steps:
       - uses: actions/checkout@v3

--- a/py-polars/requirements-dev.txt
+++ b/py-polars/requirements-dev.txt
@@ -13,7 +13,7 @@ xlsx2csv
 # Tooling
 hypothesis==6.54.5
 maturin==0.13.2
-mypy==0.971
+mypy==0.981
 pytest==7.1.3
 pytest-cov==3.0.0
 

